python/social_bot/envs/embodied_teacher.py

```diff
@@ -0,0 +1,265 @@
+                pioneer2dx_noplugin, turtlebot, youbot_noplugin for now
+                note that 'agent_type' should be the same str as the model's name
+            tasks (list): a list of teacher.Task, e.g., GoalTask, KickingBallTask
+            with_language (bool): The observation will be a dict with an extra sentence
+            use_image_observation (bool): Use image, or use low-dimentional states as
+                observation. Poses in the states observation are in world coordinate
+            image_with_internal_states (bool): If true, the agent's self internal states
+                i.e., joint position and velocities would be available together with image.
+                Only affect if use_image_observation is true
+            world_time_precision (float|None): this parameter depends on the agent. 
+                if not none, the default time precision of simulator, i.e., the max_step_size
+                defined in the agent cfg file, will be override. Note that pr2 and iCub
+                requires a max_step_size <= 0.001, otherwise cannot train a successful policy.
+            step_time (float): the peroid of one step() function of the environment in simulation.
+                step_time is rounded to multiples of world_time_precision
+                step_time / world_time_precision is how many simulator substeps during one
+                environment step. for the tasks need higher control frequency (such as the 
+                tasks need walking by 2 legs), using a smaller step_time like 0.05 is better.
+                experiments show that iCub can not learn how to walk in a 0.1 step_time
+            real_time_update_rate (int): max update_rate per seconds. there is no limit if
+                this is set to 0. if 1:1 real time is prefered(like playing or recording video),
+                this should be set to 1.0/world_time_precision.
+            port: Gazebo port, need to specify when run multiple environment in parallel
+            action_cost (float): Add an extra action cost to reward, which helps to train
+                an energy/forces efficency policy or reduce unnecessary movements
+            resized_image_size (None|tuple): If None, use the original image size
+                from the camera. Otherwise, the original image will be resized
+                to (width, height)
+            vocab_sequence_length (int): the length of encoded sequence
+            is_training_for_teacher_policy (bool): if true this env is acting like a normal
+                gym environment without teacher, used for training the teacher policy
+            demo_by_human (bool): demo by human or by existing teacher policy
+        """
+        self._is_training_for_teacher_policy = is_training_for_teacher_policy
+        self._demo_by_human = demo_by_human and (
+            not is_training_for_teacher_policy)
+        if self._is_training_for_teacher_policy:
```

Comment:
Perhaps we don't need to consider this case in this class, because it can be handled by the current PlayGround.